mllib/src/main/scala/org/apache/spark/ml/r/ALSWrapper.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.ml.r
+
+import org.apache.hadoop.fs.Path
+import org.json4s._
+import org.json4s.JsonDSL._
+import org.json4s.jackson.JsonMethods._
+
+import org.apache.spark.ml.recommendation.{ALS, ALSModel}
+import org.apache.spark.ml.util._
+import org.apache.spark.sql.{DataFrame, Dataset}
+
+private[r] class ALSWrapper private (
+    val alsm: ALSModel,
+    val rRatingCol: String,
+    val rUserCol: String,
+    val rItemCol: String,
+    val rRegParam: Double,
+    val rMaxIter: Int) extends MLWritable {
+
+  lazy val rUserFactors: DataFrame = alsm.userFactors
+
+  lazy val rItemFactors: DataFrame = alsm.itemFactors
+
+  lazy val rRank: Int = alsm.rank
+
+  def transform(dataset: Dataset[_]): DataFrame = {
+    alsm.transform(dataset)
+  }
+
+  override def write: MLWriter = new ALSWrapper.ALSWrapperWriter(this)
+}
+
+private[r] object ALSWrapper extends MLReadable[ALSWrapper] {
+
+  def fit(data: DataFrame, features: Array[String], rank: Int, regParam: Double, maxIter: Int,
+          implicitPrefs: Boolean, alpha: Double, nonnegative: Boolean,
+          distParams: Array[Int]): ALSWrapper = {
```

Comment:
Why not using explicit params? If this is to avoid style check, you can turn it off by `// stylecheck:off`